Adds "deployable" items

Rukofamicom opened the pull request:
## About The Pull Request

Original checklist here:
<details>

- [x] Adds a new item subtype `/obj/item/deployable` which is heavily but not wholly derived from how body bags work
- [x] Refactors body bags to be a subtype of `deployable`, with a good chunk of their code moved into the parent type for other items to utilize
- [x] Adds bluespace storage capsules - a storage apparatus which can hold most of the same items* as an ordinary crate, with all the convenience of fitting in your pocket! (*Does not hold items that are bulky or larger, and does not hold most living beings)
- [x] Make rollerbeds a subtype of `deployable`, with extra attention to cyborg roller beds to ensure they don't break
- [x] Refactor barrier grenades into a subtype of deployable, with a lengthy delay on being picked back up.
-  ~~Look at how everything else within `code\game\machinery\deployable.dm` can be adapted to a similar system~~ 
- [x] *Sandbags and Wooden Planks should remain as they are, but barricades made from them should be possible to pick back up into their requisite materials*
- [x] Deployable bots, such as securitrons, medibots and such. Bots will need to be deployed in an unpowered state and undergo at least a moderate startup sequence before activating to prevent dropping an instant-beepsky, or making pocket medibots too meta for healing on the go.
- [x] Sprites for bluespace storage capsules
- [x] Sounds for bluespace storage capsules
- [x] Change sprite for inactive security barriers
- [x] Add a check for deployables in dense spaces
- [x] Prevent picking up damaged barricades or transfer damage to the carried version
- [x] Add bluespace storage capsules to applied bluespace research
- [x] Block bluespace capsules from being deployed by anything other than their innate timer

</details>

### Back-end changes and minor notes
* Adds a new item subtype `/obj/item/deployable` which is derived from how body bags and roller beds work
* Moves Body bags, roller beds (including cyborg), and security barrier grenades to this new item type, and cleans up a lot of unnecessary code in the process. 
  * Body bags and roller beds should have no functional difference, except that cyborgs got a new sprite for an "empty" roller bed slot.
* Adds a check to locker logic so that they can actually be set to a mob capacity of zero - previously the loop would run once even if capacity was zero and not catch that it wasn't supposed to store any mobs at all until one had already been stored.
* Adds a bounty for bluespace capsules

### Bluespace capsules!
![Bluespace_Locker (1)](https://user-images.githubusercontent.com/9547572/236960977-68d50ecd-21f2-4785-aa1e-8cf02c7af60b.gif)

* In compressed form, a tiny item that fits in your pocket!
* In expanded form, has almost all of the storage capacity of a crate!
* Is not able to hold items of bulky size or larger, and cannot hold most creatures (Creatures the size of a mouse will fit)
* Available as a part of the Applied Bluespace research node, they take Iron, titanium and a bit of diamond and bluespace crystal to make.

### Carryable bots!
![dreamseeker_aVVXQFGkWX](https://user-images.githubusercontent.com/9547572/236961549-bf3854c3-68be-400b-b9ab-acd0b931ad32.gif)

* A new "phantom" item has been added that acts as a sort of holder for bots
* Power down a robot (requires ID) and drag the robot onto yourself to pick them up as a bulky item
* Click anywhere adjacent to place the robot back down, and interact with them normally to power it back on (Don't forget to swipe your ID again to lock its controls!)
* Powering on robots now has a brief startup sequence to prevent a couple of painful cheese strats this would have otherwise enabled

### Security Barriers are grenades no more!
 
![image](https://user-images.githubusercontent.com/9547572/236961130-73fcd3dd-be0a-434f-8cdb-5dc9bda03663.png)

* Security barriers are now possible to break down into a carryable item (which is a subtype of `obj/item/deployable`) with the swipe of a security ID for relocation
* Folded barriers are a small item and do not require security clearance to deploy
* Expanded barriers are unchanged beyond the ability to break them down
* Barriers are now always deployed one at a time, but the roundstart supply and cargo order size has been increased to compensate for this change.

### Other barricades can now be relocated!
 
* Sandbags and Wooden barricades can now be recollected as their base items (**not** refactored into `obj/item/deployable`)
* Sandbags have a chance proportional to their damage % to fall apart when recollected due to holes in the bags causing them to spill out
* Wooden barricades will return only a fraction of planks depending on the condition they're in. Old shoddy barricades that spawn as parts of some mapping areas will automatically give less planks even if not damaged from their starting values. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

* Re-introduces some of the better unintended uses for bluespace body bags (soft removed in #8973) in both a more balanced and immersive way, minimizing the unnecessary splash damage to cool gameplay on an otherwise necessary removal. 
* Gives items which already behave somewhat similarly a common parent type
* Improves immersion and strategic gameplay by making it possible to relocate barricades as needs arise (in the few cases where barricades are utilized anyway). 
* It opens the door to some cool tech-oriented items that flavorfully fit the setting really well, even beyond the concepts I currently listed, such as deployable turrets and perhaps even shuttles with some extra work.  

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>

### Bluespace Capsule:

**Research:**
![image](https://user-images.githubusercontent.com/9547572/236956399-6fee8d06-6532-40cd-bf69-b25745158879.png)

**Printing from lathe:**
![image](https://user-images.githubusercontent.com/9547572/236956719-5c3e3cb5-6d60-41d8-8d14-2f5541e67b7d.png)

**Capsule demonstration with sounds**
https://user-images.githubusercontent.com/9547572/236956810-1f17e65f-4a88-43e9-bb7a-86b64cc047e8.mp4

**Practical demonstration** (Note that the bag is too large for the capsule due to being bulky)
![dreamseeker_USQx4scAAB](https://user-images.githubusercontent.com/9547572/236956845-7768dcea-c400-4f08-9748-b7de6467b486.gif)

**In-hand sprite for capsule with button clicking down**
![dreamseeker_BOv6x2xNbL](https://user-images.githubusercontent.com/9547572/236957048-fb769845-f16f-478d-94aa-bf2b7ac5034b.gif)

### Roller Beds:

**Player functionality with roller bed**
![dreamseeker_l2ALI4vddU](https://user-images.githubusercontent.com/9547572/236956965-8d34658f-cb50-4ca6-8e75-cdad86964763.gif)

**Cyborg functionality with roller bed dispenser**
![dreamseeker_gYDhr7cM0i](https://user-images.githubusercontent.com/9547572/236956917-5a63cd39-772b-40b9-8087-553184cc1b22.gif)

**New sprite for cyborg roller bed loader when it is empty**
![dreamseeker_MgNXpxeEzx](https://user-images.githubusercontent.com/9547572/236957000-38f372df-1f9d-4d06-b356-de68f7982738.gif)

### Barricades:

**Security Barricade, including new sprite for inactive barricades:**
![dreamseeker_f3Q5Pp6sAc](https://user-images.githubusercontent.com/9547572/236957129-45ba505f-1595-420c-b3ec-571cc78f54d9.gif)

![dreamseeker_4hFIDYIC2a](https://user-images.githubusercontent.com/9547572/236957166-da3a42a2-34e4-4a97-979b-2d1319932b44.gif)

(Not shown - testing the ID swiping functionality, it works)

**Wooden and Sandbag barricades picked up:**

![dreamseeker_5T9wLAOWG8](https://user-images.githubusercontent.com/9547572/236957532-f341d829-983e-4a20-8fb6-682e13234198.gif)
![image](https://user-images.githubusercontent.com/9547572/236957545-24324361-0f91-4f3f-9518-3b0781249fe5.png)

![dreamseeker_6RZAWcI44Q](https://user-images.githubusercontent.com/9547572/236957682-0f97c22e-6b91-4cd9-9474-dda0e57ccf5f.gif)
![image](https://user-images.githubusercontent.com/9547572/236957707-a956864e-ce78-4609-a9d0-d5d5b630a3b1.png)

The amount of salvaged planks scales with the amount of damage the barricade has correctly.

### Deployable Bots

**Pick-up messages:**
![image](https://user-images.githubusercontent.com/9547572/236958004-90ec9432-424d-4781-a640-29ce90501839.png)
![image](https://user-images.githubusercontent.com/9547572/236958022-5f5efa4d-0419-4550-baf3-81341c5d6232.png)

**Actual deployments and boot times**
![dreamseeker_aVVXQFGkWX](https://user-images.githubusercontent.com/9547572/236958316-16cf12ba-08fd-41ff-812e-454ef2f8624d.gif)

![dreamseeker_xRdrirCpp6](https://user-images.githubusercontent.com/9547572/236958329-7fcff49a-cb42-4572-abbb-55c3ea8ffb88.gif)

</details>

## Changelog
:cl: Rukofamicom, Pirill, Naevi
add: Added the bluespace capsule, available through applied bluespace research. Press the button and toss the capsule and watch as it expands into a tall storage device. Close it to shrink it back into a pocket-sized capsule. Has the same capacity as a normal crate but cannot hold bulky items or most mobs (Tiny mobs like mice will fit inside).
add: Added the ability to pick up most bots such as beepsky and medibots by powering them down (requires ID access) and then dragging them onto yourself. Mulebots and ED209 are too large to be carried and do not work with this functionality.
tweak: All bots now have a 4-8 second startup sequence when powered on to prevent very sudden surprise attacks utilizing bot deployments. 
code: Adds a new object type: "/obj/item/deployable" for items which can be toggled between a deployable form that is carried around and "deployed" form that is interactable within the environment
refactor: Refactored body bags and roller beds to be subtypes of deployable mentioned above. There should be no functional difference in how these items behave. 
refactor: Refactored barrier grenades to be a subtype of deployable - these are no longer grenades and are instead collapsible barriers now. The barriers may only be dismantled after being unlocked by swiping an ID with security clearance.
tweak: The amount of security barriers available at roundstart and within cargo orders has been roughly doubled because they are now always deployed one.
tweak: Sandbags and wooden barricades can now be recollected as well by dragging them onto the user's sprite. Damage sustained to these barricades affects their return yield when picked up. 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rukofamicom committed:
Initial commit

Rukofamicom committed:
icons for capsule crates

Rukofamicom committed:
Functional storage capsules

Rukofamicom committed:
Polishing the capsule

Still needs an opening sound

Rukofamicom committed:
Finishing up capsules

Rukofamicom committed:
refactors roller beds under deployable

Rukofamicom committed:
Security barricades are now deployable

Initial commit, still needs some polish

Rukofamicom committed:
Sandbags and Wooden barricades can now be dismantled

Rukofamicom committed:
Bots are now deployable

Rukofamicom committed:
Final touches on capsule

Rukofamicom committed:
Adds capsule to applied bluespace research

Also adds it to science bounties.

Rukofamicom committed:
Cleaning up

Rukofamicom committed:
Update bot.dm

Bots are now bulky

Rukofamicom committed:
Update code/game/objects/items/deployable/bodybag.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

Rukofamicom committed:
resolve conflict

Rukofamicom committed:
Merge branch 'deployables' of https://github.com/Rukofamicom/BeeStation-Hornet into deployables

Rukofamicom committed:
Update barricade.dm

Rukofamicom committed:
Merge remote-tracking branch 'upstream/master' into deployables

Rukofamicom committed:
Updates radstation barrier grenades

Rukofamicom committed:
Updates radstation rollerbed

Rukofamicom committed:
Merge branch 'master' into deployables

Rukofamicom committed:
Bot holder updated

Bots are now carried as the contents of the bot holder. Still need to make the holder self-destruct if the bot is pulled out by some means other than deployment such as being yoinked by Xenobio extract or otherwise.

Rukofamicom committed:
Bot holder now attempts to destroy itself if the bot is removed

also adds a couple of fingerprinting lines

Rukofamicom committed:
Merge remote-tracking branch 'upstream/master' into deployables

Rukofamicom committed:
Re-updates icons

PowerfulBacon commented on code/game/objects/items/deployable/barricade.dm:
```diff
@@ -0,0 +1,212 @@
+	///How many materials are returned if the barricade is picked up at full integrity
+	var/drop_amount = 1
+
+/obj/structure/barricade/deconstruct(disassembled = TRUE)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		make_debris()
+	qdel(src)
```
Good use of deconstruct instead of Destroy() :+1:

PowerfulBacon commented on code/game/objects/items/deployable/barricade.dm:
```diff
@@ -0,0 +1,212 @@
+
+			usr.visible_message("<span class='notice'>[usr] picks up [src].</span>", "<span class='notice'>You pick up [src].</span>")
+			qdel(src)
+			return TRUE
```
I think rather than returning true/false you should have an overridable proc that creates the pickup item

Rukofamicom replied:
Done now 👍

PowerfulBacon commented on code/game/objects/items/deployable/barricade.dm:
```diff
@@ -0,0 +1,212 @@
+	if(..())
+		var/obj/item/deployable/barricade/security/carryable = new(loc)
+		usr.put_in_hands(carryable)
+		qdel(src)
```
Qdel while qdelling?

Rukofamicom replied:
uh whoops

PowerfulBacon commented on code/game/objects/items/deployable/bot.dm:
```diff
@@ -0,0 +1,15 @@
+
+/obj/item/deployable/bot/proc/self_destruct()
+	SIGNAL_HANDLER
+	for(var/atom/movable/A in contents)
+		return
```
```suggestion
	if (locate(/atom/movable) in contents)
		return
```

PowerfulBacon commented on code/game/objects/items/deployable/deployable.dm:
```diff
@@ -0,0 +1,97 @@
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	to_chat(user, "<span class='notice'>You try to force your way out of [src]...</span>")
+	to_chat(loc, "<span class='warning'>Someone starts trying to break free of [src]!</span>")
```
to_chat loc? What is the intention behind this (Does this work)?
If you want nearby people to see it, use visible_message instead

Rukofamicom replied:
That is a good question tbh.

This is taken from existing code as-is iirc so I didn't test it. Will adjust to proper code.

Rukofamicom replied:
To answer the question, yes it does work and it is actually pretty frequently used too.
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/9547572/d3732d53-81fc-47b8-8f1d-4b922a49e8f3)

I know **why** it is used now that I'm not reading this out of context too - so the message is **only** visible to people on the same tile. In this case, only visible to the person holding the item. With the removal of the bluespace bodybag this is actually unused code now because there's no way for players to be trapped inside of deployables, so I will just completely remove this entire proc to clean the code up.

PowerfulBacon commented on code/game/objects/items/deployable/rollerbed.dm:
```diff
@@ -0,0 +1,26 @@
+	return ..()
+
+/obj/item/deployable/rollerbed/robo/afterattack(atom/target, mob/user, proximity)
+	if(istype(target, /obj/structure/bed/roller/))
```
```suggestion
	if(istype(target, /obj/structure/bed/roller))
```

PowerfulBacon reviewed (changes requested):
No issues otherwise

Rukofamicom committed:
Update code/game/objects/items/deployable/rollerbed.dm

Remove unnecessary slash

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

Rukofamicom committed:
No more qdel during qdel

qdelception is bad or something probably.

Rukofamicom committed:
Removes container_resist()

It is not possible for mobs to be trapped within deployables by design. With the removal of bluespace bodybags I don't foresee this code being used again.

Rukofamicom committed:
Update barricade.dm

Rukofamicom committed:
Removes FALSE returns

These just need to break away from the current proc, the value is of no use anymore.
